packages/flutter/lib/src/gestures/monodrag.dart

```diff
@@ -127,7 +127,10 @@ abstract class DragGestureRecognizer extends OneSequenceGestureRecognizer {
     if (event is PointerMoveEvent) {
       final VelocityTracker tracker = _velocityTrackers[event.pointer];
       assert(tracker != null);
-      tracker.addPosition(event.timeStamp, event.position);
+      // PointerMoveEvent from platform's gesture up events are positionally redundant
+      // vs the last PointerMoveEvent. Do not use in velocity calculation.
+      if (event.sourceEventType != PointerChange.up)
+        tracker.addPosition(event.timeStamp, event.position);
```

Comment:
ya, it's not really a move event as in the user still had the finger touching the screen at 1,1 at time x+32ms. The user stopped touching the screen at x+16ms. 

On Android, the sequence would have been PointerMoveEvent -> PointerMoveEvent -> PointerUpEvent which is semantically easier to parse than iOS's PointerMoveEvent -> PointerMoveEvent -> PointerMoveEvent where the last PointerMoveEvent was really a PointerUpEvent from the OS.